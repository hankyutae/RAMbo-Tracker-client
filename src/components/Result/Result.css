#result-list {
    /* background-color: rgb(131, 120, 103); */
    background-color: rgb(204, 204, 204);
    border-radius: 5px;
    margin: 10px;
    font-size: 20px;
    font-weight: 600;
    color: black;
    line-height: 1.7;
}

li {
    border-radius: 10px;
    margin: 10px;
}

ul {
    padding-inline-start: 0px;
}
/*^ fixes issue with default browser styling*/

.results {
    display: block;
    width: 50%;
}

.pie-area {
    text-align: center;
}

#results-button{
    padding: 10px;
    text-align: center;
}

#results-button button{
    border-radius: 5px;
    border: 1px solid white;
    margin: 0 auto;
    padding: 15px;
    background-color: #32936F;
    color: white;
}
#results-button button:hover{
    background-color: rgb(31, 97, 73)
}
@media screen and (max-width: 500px){

    #results-button button{
        width: 40%;
        padding: 5px;
    }

}