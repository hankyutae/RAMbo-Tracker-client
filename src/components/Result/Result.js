import React from 'react';
import Api from '../../services/api-service';
import PieChart from '../PieChart/PieChart';
import './Result.css';

export default class Result extends React.Component {
  state = {
    results: null,
    error: null
  };
  refreshResults = () => {
    Api.doFetch("/results")
      .then(res => {
        this.setState({ results: res });
      })
      .catch(res=>
        this.setState({error: res.message}));
  };
  render() {
    let results = !this.state.results ? (
      <h2>Click Analyze to See Results</h2>
    ) : (
      this.state.results.map((item, key) => {
        let data = [];
        return (
          <li className="visual-list" id="result-list" key={key}>
            <strong>{item.symptomType.type}</strong> is experienced most
            frequently after eating foods with:{' '}
            <i>
              {item.mostCommonIngredients
                .filter((food) => {
                  return food.name !== 'WATER';
                })
                .map((food) => {
                  data.push({
                    label: food.name.toLowerCase(),
                    value: food.weight
                  })
                  return food.name.toLowerCase();
                })
                .join(', ')}
            </i>
            <div className='pie-area'>
            <PieChart
              data={data}
              title={item.symptomType.type}
              colors={[
                '#8AD2D8',
                '#C6A68E',
                '#558AA4',
                '#F15E3D',
                '#56704B',
                '#CC3B7C',
                '#005594',
                '#89B65A',
                '#EBC9BE',
                '#EFCBE'

              ]}
            />
            </div>
            
          </li>
        );
      })
    );
    return (
      
      <div id="results">
        <h2>My Results</h2>
{this.state.error && <p class="error">There Was An Error!</p>}
        <ul>{results}</ul>
        
        <div id="results-button">
          <button onClick={e => this.refreshResults(e)}>Refresh Results</button>
        </div>
      </div>
    );
  }
}
