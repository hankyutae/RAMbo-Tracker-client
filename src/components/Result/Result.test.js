import React from 'react';
import ReactDOM from 'react-dom';
import Result from './Result';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Result/>
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<BrowserRouter>
      <Result/>
    </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});