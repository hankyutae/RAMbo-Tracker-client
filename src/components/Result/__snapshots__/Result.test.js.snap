// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="results"
>
  <ul>
    <p>
      click refresh to see correlation
    </p>
  </ul>
  <button
    onClick={[Function]}
  >
    refresh
  </button>
</div>
`;
